integrationTests/src/test/java/org/sonarqube/gradle/AbstractGradleIT.java

```diff
@@ -67,6 +73,90 @@ protected static Semver getGradleVersion() {
+    if (absoluteProjectBaseDir == null) {
+      throw new IllegalStateException("sonar.projectBaseDir is null");
+    }
+    Set<String> hiddenProperties = new HashSet<>(Arrays.asList(
+      // different on each machine, nothing to compare
+      "sonar.java.jdkHome", "sonar.scanner.arch", "sonar.scanner.os", "sonar.scanner.opts",
+      // path to a different temporary file for each run, e.g. junit17577551948278219342.tmp
+      "sonar.scanner.internal.dumpToFile",
+      // depends on the plugin and gradle version that run the test, e.g. 6.3-SNAPSHOT/Gradle 9.0.0
+      "sonar.scanner.appVersion"));
+    // All environment variables started with SONAR_SCANNER_* will pollute the test by adding sonar.scanner.* properties
+    // We are forced to filter them out
+    Set<String> testedSonarScannerProperties = new HashSet<>(Arrays.asList(
+      "sonar.scanner.apiBaseUrl",
+      "sonar.scanner.app",
+      "sonar.scanner.appVersion",
+      "sonar.scanner.arch",
+      "sonar.scanner.internal.dumpToFile",
+      "sonar.scanner.os",
+      "sonar.scanner.wasEngineCacheHit"));
+    Map<String, String> replacementMap = new LinkedHashMap<>();
+    replacementMap.put("${parentBaseDir}", Paths.get(absoluteProjectBaseDir).getParent().toString());
+    replacementMap.put("${PWD}", System.getProperty("user.dir"));
+    replacementMap.put("${HOME}", System.getProperty("user.home"));
+    String cirrusWorkingDir = System.getenv("CIRRUS_WORKING_DIR");
+    if (cirrusWorkingDir != null) {
+      replacementMap.put("${HOME}", cirrusWorkingDir);
+    }
+    Pattern dependenciesInGradleCache = Pattern.compile("(?<=,|^)/[^,]+/.gradle/caches/modules-\\d+/files-[0-9.]+/" +
+      "(?<groupId>[^/,]++)/(?<artifactId>[^/,]++)/(?<version>[^/,]++)/[0-9a-f]{40}/\\k<artifactId>-\\k<version>\\.jar(?=,|$)");
+    props.stringPropertyNames().stream()
+      .sorted()
+      .filter(key -> !key.startsWith("sonar.scanner.") || testedSonarScannerProperties.contains(key))
+      .forEach(key -> {
+        String value = props.getProperty(key);
+        if (hiddenProperties.contains(key)) {
+          value = "<hidden>";
+        }
+        for (Map.Entry<String, String> entry : replacementMap.entrySet()) {
+          value = removeAbsolutePath(entry.getKey(), entry.getValue(), value);
+        }
+        value = value.replace('\\', '/');
+        for (String replacementKey : replacementMap.keySet()) {
+          value = removeAbsolutePath("${M2}", replacementKey + "/.m2", value);
+        }
+        value = dependenciesInGradleCache.matcher(value).replaceAll(res ->
+        // Java 11 does not support res.group("groupId") this is why there is res.group(1)
+        "\\${M2}/repository/" + res.group(1).replace(".", "/") + "/${artifactId}/${version}/${artifactId}-${version}.jar");
+        result.put(key, value);
+      });
+    return result;
+  }
+
+  private static String removeAbsolutePath(String placeHolder, String absolutePath, String value) {
```

Comment:
Nitpick: The method does not seem to be aware that parameters are path, can we rename this to `replaceAllPatterns(String placeHolder, String pattern, String value)` ?